Lib/test/string_tests.py

```diff
@@ -1090,11 +1090,14 @@ def test_py3x_warnings_isinstance(self):
             isinstance("fix", basestring)
 
     def test_py3x_warnings_join(self):
-        with test_support.check_py3k_warnings(("string concatenation on mixed types is not supported in 3.x: "
-                                  "encode or decode the strings to match types", Py3xWarning)):
+        with test_support.check_py3k_warnings(("mixed bytes, str and unicode operands cannot be used in string concatenation in Python 3.x"
```

Comment:
The way these strings are concatenated, won't we end up with: "concatenation in Python 3.xconvert the..."?

Maybe the colon was correct here? Not sure.